### ISSUE TITLE
Parameters in policy initiative "Configure Azure PaaS services to use private DNS zones" is empty

### ISSUE THREAD

**Issue by tlfzhylj** [Azure/terraform-azurerm-caf-enterprise-scale#706]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.4.4

**azure provider**: 3.50.0

**module**: 3.3.0

#### Describe the bug

This is partly related to #705 

I can't get private DNS zones and private endpoints to work. I've deployed some private endpoints in a subscription under the Corp management group, and all the policies are enabled. Also the policy named "Configure Azure PaaS services to use private DNS zones".

As you can see in the screenshot, no DNS zones are set in the policy.

I can imagine this is because I have split the core, connectivity and management in three different Terraform projects - much like described in [[Examples] Deploy using multiple module declarations with remote state](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/wiki/%5BExamples%5D-Deploy-using-multiple-module-declarations-with-remote-state).

How can this best be solved?

Thanks again!

#### Screenshots
![image](https://user-images.githubusercontent.com/29739663/236552730-0f7289b4-30fd-49aa-8c22-bb09d9105493.png)

**Comment by tlfzhylj** [comment]
I found out how to fix this.

I had to configure my custom resource group name when calling the core module.

```hcl
module "core" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "4.0.1"

  # (...) All other properties removed from example

  configure_connectivity_resources = {

    # Advanced configuration
    # Setting custom names because module does not follow naming standards
    advanced = {
      custom_settings_by_resource_type = {
        azurerm_resource_group = {
          dns = {
            (local.default_location) = {
              name = "<my custom resource group name>"
            }
          }
        }
      }
    }
  }
}

```

**Comment by tlfzhylj** [comment]
@matt-FFFFFF 
Regarding the same policy, why is the enforcementMode defaulting to false for this policy?

I had to set the following for the policy to actually work, and not just show the compliance:

```hcl
  archetype_config_overrides = {
    <myorg>-corp = {
      enforcement_mode = {
        Deploy-Private-DNS-Zones = true
      }
    }
  }
```

**Comment by tlfzhylj** [comment]
The original issue was solved with version 4.0.2 of the module.

**Comment by tlfzhylj** [comment]
> @matt-FFFFFF Regarding the same policy, why is the enforcementMode defaulting to false for this policy?
> 
> I had to set the following for the policy to actually work, and not just show the compliance:
> 
> ```terraform
>   archetype_config_overrides = {
>     <myorg>-corp = {
>       enforcement_mode = {
>         Deploy-Private-DNS-Zones = true
>       }
>     }
>   }
> ```

I registered a new issue regarding this: #721
